packages/components/pie-card-container/test/visual/pie-card-container.spec.ts

```diff
@@ -16,15 +34,48 @@ const slotContent = `<div style="font-size: calc(var(--dt-font-body-l-size) * 1p
     tempora asperiores aspernatur atque quas.</p>
 </div>`;
 
-test.describe('PieCardContainer - Visual tests`', () => {
-    test('should display the PieCardContainer component slot content successfully', async ({ page, mount }) => {
-        await mount(PieCardContainer, {
-            props: {} as CardContainerProps,
-            slots: {
-                default: slotContent,
-            },
-        });
-
-        await percySnapshot(page, 'PieCardContainer - Visual Test');
+// Renders a <pie-card-container> HTML string with the given prop values
+const renderTestPieCardContainer = (propVals: WebComponentPropValues) => `<pie-card-container variant="${propVals.variant}" ${propVals.disabled ? 'disabled' : ''} >${slotContent}</pie-card-container>`;
+
+const componentPropsMatrix : WebComponentPropValues[] = getAllPropCombinations(props);
+const componentPropsMatrixByVariant: Record<string, WebComponentPropValues[]> = splitCombinationsByPropertyValue(componentPropsMatrix, 'variant');
+const componentVariants: string[] = Object.keys(componentPropsMatrixByVariant);
+
+// This ensures the component is registered in the DOM for each test
+// This is not required if your tests mount the web component directly in the tests
+test.beforeEach(async ({ page, mount }) => {
+    await mount(
+        PieCardContainer,
+        {},
+    );
+
+    // Removing the element so it's not present in the tests (but is still registered in the DOM)
+    await page.evaluate(() => {
+        const element : Element | null = document.querySelector('pie-card-container');
+        element?.remove();
     });
 });
+
+test.describe('PieCardContainer - Visual tests`', () => {
+    componentVariants.forEach((variant) => test(`Render all prop variations for Variant: ${variant}`, async ({ page, mount }) => {
```

Comment:
Please could you change this to:

`should render all prop variations for Variant: ....`

Next PR is ok

Reply:
added it here https://github.com/justeattakeaway/pie/pull/791